packages/commons-test/src/testUtils.ts

```diff
@@ -0,0 +1,149 @@
+  state: descriptorState.published,
+  attributes: {
+    certified: certifiedAttributes,
+    declared: declaredAttributes,
+    verified: verifiedAttributes,
+  },
+});
+
+export const buildEServiceAttribute = (
+  attributeId: AttributeId = generateId<AttributeId>()
+): EServiceAttribute => ({
+  ...generateMock(EServiceAttribute),
+  id: attributeId,
+});
+
+export const buildEServiceAttributes = (num: number): EServiceAttribute[] =>
+  new Array(num).map(() => buildEServiceAttribute());
+
+export const buildEService = (
+  eserviceId: EServiceId = generateId<EServiceId>(),
+  producerId: TenantId = generateId<TenantId>(),
+  descriptors: Descriptor[] = []
+): EService => ({
+  ...generateMock(EService),
+  id: eserviceId,
+  producerId,
+  descriptors,
+});
+
+export const buildVerifiedTenantAttribute = (
+  attributeId: AttributeId = generateId<AttributeId>()
+): TenantAttribute => ({
+  ...generateMock(VerifiedTenantAttribute),
+  id: attributeId,
+});
+
+export const buildVerifiedTenantAttributes = (num: number): TenantAttribute[] =>
+  new Array(num).map(() => buildVerifiedTenantAttribute());
+
+export const buildCertifiedTenantAttribute = (
+  attributeId: AttributeId = generateId<AttributeId>()
+): CertifiedTenantAttribute => ({
+  ...generateMock(CertifiedTenantAttribute),
+  id: attributeId,
+  type: tenantAttributeType.CERTIFIED,
+  revocationTimestamp: undefined,
+});
+
+export const buildCertifiedTenantAttributes = (
+  num: number
+): TenantAttribute[] =>
+  new Array(num).map(() => buildCertifiedTenantAttribute());
+
+export const buildDeclaredTenantAttribute = (
+  attributeId: AttributeId = generateId<AttributeId>()
+): TenantAttribute => ({
+  ...generateMock(DeclaredTenantAttribute),
+  id: attributeId,
+});
+
+export const buildDeclaredTenantAttributes = (num: number): TenantAttribute[] =>
+  new Array(num).map(() => buildDeclaredTenantAttribute());
+
+export const buildTenant = (
+  tenantId: TenantId = generateId<TenantId>(),
+  attributes: TenantAttribute[] = []
+): Tenant => ({
+  ...generateMock(Tenant),
+  id: tenantId,
+  externalId: {
+    value: uuidv4(),
+    origin: "EXT",
+  },
+  attributes,
+});
+
+export const buildAgreementWithValidCreationState = (
+  eserviceId: EServiceId = generateId<EServiceId>(),
+  consumerId: TenantId = generateId<TenantId>()
+): Agreement => {
+  // Please before change this for some reason, check agreementCreationConflictingStates in packages/agreement-process/src/model/domain/models.ts
+  const agreementCreationConflictingStates: AgreementState[] = [
+    agreementState.draft,
+    agreementState.pending,
+    agreementState.missingCertifiedAttributes,
+    agreementState.active,
+    agreementState.suspended,
+  ];
+
+  return {
+    ...generateMock(Agreement),
+    eserviceId,
+    consumerId,
+    state: randomArrayItem(
+      Object.values(agreementState).filter(
+        (state) => !agreementCreationConflictingStates.includes(state)
+      )
+    ),
+  };
+};
```

Comment:
why not just
```suggestion
export const buildAgreement = (
  eserviceId: EServiceId = generateId<EServiceId>(),
  consumerId: TenantId = generateId<TenantId>(),
  state: AgreementState
): Agreement => {
  return {
    ...generateMock(Agreement),
    eserviceId,
    consumerId,
    state,
  };
};
```
We can both avoid uncheckable dependencies and make the test more clear

Reply:
For sure it's more redeable, using this way we need to provide a state as parameter and we forced to write the same code around the tests with a `randomArrayItem` for "Archived" and "Rejected" states usages.
This is the main reason behind this choice, in a few words we favor to add, much as possibile the code in utils methods rather than repeat operation around tests, but this approach make difficult to binding the util's method with `agreementCreationConflictingStates` (it introduces a circular dependancies).
Anyway at the moment there are a few point where this method are used, so I have done the changes and we will evaluate in future if necessary some refactor to avoid the eventually boiler code.